src/openms/include/OpenMS/KERNEL/RangeManager.h

```diff
@@ -134,14 +145,68 @@ namespace OpenMS
       min_ = std::min(min_, other.min_);
       max_ = std::max(max_, other.max_);
     }
-
+    
     /// extend the range such that it includes the given @p value
     void extend(const double value)
     {
       min_ = std::min(min_, value);
       max_ = std::max(max_, value);
     }
 
+    /// Extend the range by @p by units to left and right
+    /// Using negative values will shrink the range. It may become empty.
+    /// Calling this on an empty range will not have any effect.
+    void extendLeftRight(const double by)
+    {
+      if (isEmpty()) return;
+      min_ -= by;
+      max_ += by;
+    }
+
+    /// Ensure the range of this does not exceed the range of @p other.
+    /// If @p other already contains() this range, nothing changes.
+    /// If this range is entirely outside the range of @p other,
+    /// the resulting range is empty.
+    /// Empty ranges are not modified.
+    /// @throw Exception::InvalidRange if @p other is empty
+    void clampTo(const RangeBase& other)
+    {
+      if (isEmpty()) return;
+      if (other.isEmpty()) throw Exception::InvalidRange(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION);
+
+      min_ = std::max(min_, other.min_);
+      max_ = std::min(max_, other.max_);
+    }
+
+    /// Move range of *this to min/max of @p sandbox, without changing the span, if possible.
```

Comment:
I don't get what this is doing from the description. e.g why min get's precedence over max during shifting